internal/awsservice/cloudformation.go

```diff
@@ -0,0 +1,75 @@
+package awsservice
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+	"github.com/google/uuid"
+	"log"
+	"time"
+)
+
+const instanceIdKey = "InstanceId"
+
+func CreateStackName(stackPrefix string) string {
+	// Create stack name
+	// Only adding the first 5 char for readability
+	// Chance of conflict is 1 in 35^5 or about 1 in 52.5 million
+	generatedUuid, err := uuid.NewUUID()
+	if err != nil {
+		log.Fatalf("Failed to create uuid error %v", err)
+	}
+	return stackPrefix + generatedUuid.String()[0:5]
+}
+
+func StartStack(ctx context.Context, stackName string, client *cloudformation.Client, templateText string, timeOutInMinutes int32, parameters []types.Parameter) {
+	log.Printf("Template text : %s", templateText)
+
+	// Create cf stack config
+	createStackInput := cloudformation.CreateStackInput{
+		StackName:        aws.String(stackName),
+		TemplateBody:     aws.String(templateText),
+		TimeoutInMinutes: aws.Int32(timeOutInMinutes),
+		Parameters:       parameters,
+	}
+
+	// Start cf stack
+	_, err := client.CreateStack(ctx, &createStackInput)
+	if err != nil {
+		log.Fatalf("Failed to create stack %v", err)
+	}
+}
+
+func DeleteStack(ctx context.Context, stackName string, client *cloudformation.Client) {
+	deleteStackInput := cloudformation.DeleteStackInput{
+		StackName: aws.String(stackName),
+	}
+	_, err := client.DeleteStack(ctx, &deleteStackInput)
+	if err != nil {
+		log.Fatalf("Could not delete stack %v", stackName)
+	}
+}
+
+func FindStackInstanceId(ctx context.Context, stackName string, client *cloudformation.Client, timeOutMinutes int) string {
+	for i := 0; i <= timeOutMinutes; i++ {
+		time.Sleep(time.Minute)
```

Comment:
When looping I prefer to start with the action and end with sleep.

Also you are sleeping before the print. The print should be first.